### PR TITLE
Add ability to run multiple folders

### PR DESCRIPTION
**Pull request by codenirvana**
**Usage:**
```console
$ newman run collection.json --folder f1 --folder f2
```

**Todo:**
- [x] Add Tests
- [x] Switch to `postman-runtime` 7.2.x
- [x] Update README

### CODE REVIEW

File: bin/util.js [hunk 1]
```diff
@@ -21,15 +21,30 @@ module.exports = {
         },
 
         /**
-         * Helper for collecting global key=value variables
+         * Helper for collecting argument passed multiple times.
+         *
+         * --folder f1 --folder f2
+         *
+         * @param {String} val - The argument value.
+         * @param {Array} memo - The array that is populated by argument values.
```
Comment (kunagpal) [inline bin/util.js]: `String[]`

**Comment by kunagpal** [comment]
@codenirvana Also add integration and CLI tests. The rest looks good 😄

**Comment by shamasis** [comment]
What test is failing?

**Comment by codenirvana** [comment]
@shamasis Fixed.

File: README.md [hunk 2]
```diff
@@ -139,7 +139,8 @@ For more details on [Reporters](#reporters) and writing your own [External Repor
   Specifies the number of times the collection has to be run when used in conjunction with iteration data file.
 
 - `--folder <name>`<br />
-  Run requests within a particular folder in a collection.
+  Run requests within a particular folder(s) in a collection. Multiple folders can be specified by using
```
Comment (kunagpal) [inline README.md]: folder/folders

File: lib/run/index.js [hunk 3]
```diff
@@ -104,6 +105,14 @@ module.exports = function (options, callback) {
             }, {});
         }
 
+        // sets entrypoint to execute if options.folder is specified.
+        if (options.folder) {
+            entrypoint = { execute: options.folder };
+
+            // uses `multipleIdOrName` lookupStrategy in case of multiple folders.
+            _.isArray(entrypoint.execute) && (entrypoint.lookupStrategy = 'multipleIdOrName');
```
Comment (kunagpal) [inline lib/run/index.js]: `const`

File: lib/run/options.js [hunk 4]
```diff
@@ -230,17 +230,23 @@ var _ = require('lodash'),
...
          *
          * @param {String[]|String} value - The list of folders to execute
          * @param {Object} options - The set of wrapped options.
          * @param {Function} callback - The callback function invoked to mark the end of the folder load routine.
          * @returns {*}
          */
         folder: function (value, options, callback) {
-            let folders = _.isString(value) ? [value] : value;
+            if (!value.length) {
+                return callback(null); // avoids empty string or array
```
Comment (kunagpal) [inline lib/run/options.js]: `null` to be removed, tests updated.

File: test/unit/run.test.js [hunk 5]
```diff
@@ -225,4 +227,50 @@ describe('run module', function () {
...
+            it('should be undefined by default', function (done) {
+                run({ collection: {} }, function (options) {
+                    expect(options).to.have.property('entrypoint').to.be.undefined;
+                    done();
+                });
+            });
+
+            it('should handle options.folder passed as string correctly', function (done) {
+                run({ collection: {}, folder: 'f1' }, function (options) {
+                    expect(options).to.have.property('entrypoint').to.eql({ execute: 'f1' });
```
Comment (kunagpal) [inline test/unit/run.test.js]: Use nested/deep clause assertion.
```js
expect({a: {b: ['x', 'y']}}).to.nested.include({'a.b[1]': 'y'});
```

File: test/unit/run.test.js [hunk 6]
```diff
@@ -225,4 +227,50 @@ describe('run module', function () {
...
+        });
+
+        after(function () {
+            runtime.Runner.restore();
+        });
+
+        describe('entrypoint', function () {
+            it('should be undefined by default', function (done) {
+                run({ collection: {} }, function (options) {
+                    expect(options).to.have.property('entrypoint').to.be.undefined;
```
Comment (kunagpal) [inline test/unit/run.test.js]: `not.be.ok`

**Comment by SarathkumarRajendran** [comment]
If i specify multiple folders, say folder1, folder2, folder3, folder4, folder5, the folders are not running in sequential mode. Can you please let me know how to run the folders multiple folders sequentially?

**Pull request merged**